Apps/WebClient/src/ClientApp/src/components/modal/covid.vue

```diff
@@ -24,104 +22,102 @@ export default class CovidModalComponent extends Vue {
         userPreference: UserPreference;
     }) => void;
     @Getter("user", { namespace: "user" }) user!: User;
+    @Getter("laboratoryOrders", { namespace: "laboratory" })
+    laboratoryOrders!: LaboratoryOrder[];
 
-    @Prop() error!: boolean;
-    @Prop({ default: false }) isLoading!: boolean;
+    @Getter("getPreference", { namespace: "user" })
+    getPreference!: (preferenceName: string) => UserPreference | undefined;
 
-    private logger!: ILogger;
+    @Prop({ default: false }) isLoading!: boolean;
 
-    private mounted() {
-        this.logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+    private isDismissed = false;
+
+    private get isVisible(): boolean {
+        if (this.isLoading || this.isDismissed) {
+            return false;
+        } else if (this.laboratoryOrders.length > 0) {
+            const preference = this.getPreference(
+                UserPreferenceType.ActionedCovidModalAt
+            );
+            if (preference != undefined) {
+                const actionedCovidModalAt = new DateWrapper(preference.value);
+                const mostRecentLabTime = new DateWrapper(
+                    this.laboratoryOrders[0].messageDateTime
+                );
+                if (actionedCovidModalAt.isAfter(mostRecentLabTime)) {
+                    return false;
+                } else {
+                    return true;
+                }
+            } else {
+                return true;
+            }
+        } else {
+            return false;
+        }
     }
 
-    public isVisible = false;
-    public show = false;
-
-    private actionCovidModal() {
-        this.logger.debug("Actioning Covid Modal...");
-        const actionedCovidPreference = UserPreferenceType.ActionedCovidModalAt;
+    private updateCovidPreference() {
+        const preferenceName = UserPreferenceType.ActionedCovidModalAt;
         let isoNow = new DateWrapper().toISO();
-        if (this.user.preferences[actionedCovidPreference] != undefined) {
-            this.user.preferences[actionedCovidPreference].value = isoNow;
+        if (this.user.preferences[preferenceName] != undefined) {
```

Comment:
when we read code, I prefer actionedCovidPreference is more descritive and provides more contexts than preferenceName.